spring-shell2-shell1-adapter/src/main/java/org/springframework/shell2/legacy/LegacyParameterResolver.java

```diff
@@ -82,7 +86,40 @@ public Object resolve(MethodParameter methodParameter, List<String> words) {
 
 	@Override
 	public ParameterDescription describe(MethodParameter parameter) {
-		throw new UnsupportedOperationException();
+		Parameter jlrParameter = parameter.getMethod().getParameters()[parameter.getParameterIndex()];
+		CliOption option = jlrParameter.getAnnotation(CliOption.class);
+		ParameterDescription result = ParameterDescription.outOf(parameter);
+		result.help(option.help());
+		result.defaultValue(defaultValueFor(option));
+		result.keys(Arrays.asList(option.key()));
+		result.mandatoryKey(false);
```

Comment:
I was wondering if this should be:
```java
result.mandatoryKey(option.mandatory())
```
I was reading the docs of ParameterDescription#mandatoryKey and I understood it to be dependant on the ParamenterResolver and not on the configuration of the parameter.